File: airesources/Java/hlt/Planet.java
package hlt;

import java.util.Collections;
import java.util.List;

public class Planet extends Entity {

    private final int remainingProduction;
    private final int currentProduction;
    private final int dockingSpots;
    private final List<Integer> dockedShips;

    public Planet(final int owner, final int id, final double xPos, final double yPos, final int health,
                  final double radius, final int dockingSpots, final int currentProduction,
                  final int remainingProduction, final List<Integer> dockedShips) {

        super(owner, id, xPos, yPos, health, radius);

        this.dockingSpots = dockingSpots;
        this.currentProduction = currentProduction;
        this.remainingProduction = remainingProduction;
        this.dockedShips = Collections.unmodifiableList(dockedShips);
    }

    public int getRemainingProduction() {
        return remainingProduction;
    }

    public int getCurrentProduction() {
        return currentProduction;
    }

    public int getDockingSpots() {
        return dockingSpots;
    }

    public List<Integer> getDockedShips() {
        return dockedShips;
    }

    public boolean isFull() {
        return dockedShips.size() == dockingSpots;
    }

    public boolean isOwned() {
        return getOwner() != -1;
    }

    @Override
    public String toString() {
        return "Planet[" +
                super.toString() +
                ", remainingProduction=" + remainingProduction +
                ", currentProduction=" + currentProduction +
                ", dockingSpots=" + dockingSpots +
                ", dockedShips=" + dockedShips +
                "]";
    }
}

File: airesources/Scala/hlt/Planet.scala
package hlt

class Planet(override val owner: Option[Short],
             override val id: Long,
             override val xPos: Double,
             override val yPos: Double,
             override val health: Short,
             override val radius: Double,
             val dockingSpots: Short,
             val currentProduction: Short,
             val remainingProduction: Short,
             val dockedShips: Seq[Long])
    extends Entity(owner, id, xPos, yPos, health, radius) {
  def isFull: Boolean = dockedShips.size == dockingSpots

  def isOwned: Boolean = owner.isDefined

  override def toString: String =
    "Planet[" + super.toString + ", remainingProduction=" + remainingProduction + ", currentProduction=" +
      currentProduction + ", dockingSpots=" + dockingSpots + ", dockedShips=" + dockedShips + "]"
}
